test/download-dist

```diff
@@ -58,66 +59,35 @@ def download_dist(wait=False):
         message("download-dist: uncommitted local changes, skipping download")
         return False
 
-    download_url = f"https://github.com/{ os.getenv('GITHUB_BASE', 'cockpit-project/cockpit-machines') }-dist/raw/main/{sha}.tar"
-    request = urllib.request.Request(download_url)
-    tario = io.BytesIO()
+    dist_git_checkout = os.path.join(os.getenv("XDG_CACHE_HOME", os.path.expanduser("~/.cache")), "cockpit-dev", REPO)
+
+    if not os.path.exists(dist_git_checkout):
+        message(f"download-dist: Creating dist cache {dist_git_checkout}")
+        # we can't directly clone a tag with clone -b, so let's always checkout main
+        # this is usually very close to what we are interested in anyway
+        subprocess.check_call(["git", "clone", "--quiet", "--depth=1", "https://github.com/" + REPO, dist_git_checkout])
```

Comment:
I think we can do better here.  Two ideas:

  1) Let's rather maintain a raw clone here (ie: `cockpit-machines-dist.git`, no checkout dir) and extract from it directly into the dist/ dir of the working tree that we're inside of

  2) Do the checkout into this directory, but at least take advantage of the files being there, and do reflink-copies. It seems we might not get those by default, possibly related to the recent complaining about sendfile() and /proc.  See https://bugs.python.org/issue37157 for example.

I think I prefer option 1, though.  Having a bunch of checked out files lying around in ~/.cache/ is kinda lame.

Reply:
Nice idea, thanks for the tip with `git archive`! Done.